modules/tensor_mechanics/src/actions/TensorMechanicsAction.C

```diff
@@ -723,102 +723,80 @@ TensorMechanicsAction::actOutputMatProp()
 
   if (_current_task == "add_material")
   {
-
     // Add output Materials
     for (auto out : _generate_output)
     {
-      std::string type;
       InputParameters params = emptyInputParameters();
 
       // RankTwoCartesianComponent
-      for (const auto & r2q : _rank_two_cartesian_component_table)
-        for (unsigned int a = 0; a < 3; ++a)
-          for (unsigned int b = 0; b < 3; ++b)
-            if (r2q.first + '_' + _component_table[a] + _component_table[b] == out)
-            {
-              type = ad_prepend + "RankTwoCartesianComponent";
-              params = _factory.getValidParams(type);
-              params.set<MaterialPropertyName>("rank_two_tensor") = _base_name + r2q.second;
-              params.set<unsigned int>("index_i") = a;
-              params.set<unsigned int>("index_j") = b;
-
-              params.applyParameters(parameters());
-              params.set<std::string>("property_name") = _base_name + out;
-            }
+      if ([&]() {
```

Comment:
the lambda here adds a closure that permits exiting out of the nested loop as soon as a match is found.

Reply:
I think it is confusing to set the parameters in the arguments to an if statement.  I think the loops should be kept outside of the if statement argument.  A bool could be set in the loop to use with the continue.